services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/serialization/ResumableFileDownloadSerializer.java

```diff
@@ -62,6 +62,11 @@ public static byte[] toJson(ResumableFileDownload download) {
                                        jsonGenerator,
                                        "s3ObjectLastModified");
         }
+        if (download.s3ObjectEtag().isPresent()) {
```

Comment:
Can we add unit tests in ResumableFileDownloadSerializerTest?

Reply:
Good call. I added coverage for that new Etag field in the existing test. Do I need to add a standalone test just for that, or is this sufficient?